digitiser-aggregator/src/frame/cache.rs

```diff
@@ -27,52 +24,64 @@ where
         }
     }
 
-    pub(crate) fn push(&mut self, digitiser_id: DigitizerId, metadata: &FrameMetadata, data: D) {
-        match self
-            .frames
-            .iter_mut()
-            .find(|frame| frame.metadata.equals_ignoring_veto_flags(metadata))
-        {
-            Some(frame) => {
-                frame.push(digitiser_id, data);
-                frame.push_veto_flags(metadata.veto_flags)
-            }
-            None => {
+    #[tracing::instrument(skip_all, fields(
+        digitiser_id = digitiser_id,
+        metadata_timestamp = metadata.timestamp.to_rfc3339(),
+        metadata_frame_number = metadata.frame_number,
+        metadata_period_number = metadata.period_number,
+        metadata_veto_flags = metadata.veto_flags,
+        metadata_protons_per_pulse = metadata.protons_per_pulse,
+        metadata_running = metadata.running
+    ))]
+    pub(crate) fn push<'a>(
```

Comment:
Not a fan of this approach, as it introduces an exhaustive search of the cache that is not required.

I don't think the strong separation of the tracing specific code is helpful here, in fact I think the readability of the code suffers if return values exist only to facilitate observability.

By its very nature, observability code does end up in more than one part of a codebase, so keeping it all in `main.rs` is an anti-pattern best avoided IMO.

Reply:
Ah true enough. I assumed we'd be unlikely to have a situation where more than two frames would be in the cache at a time. I suppose that could happen if a digitiser event message goes missing